### ISSUE TITLE
oss-fuzz reports a possible write-after-free in libjxl

### ISSUE THREAD

**Issue by jcupitt** [libjxl/libjxl#66]
oss-fuzz might have found a nasty write-after-free bug in git master libjxl. This file:

http://www.rollthepotato.net/~john/.clusterfuzz-testcase-minimized-jpegsave_file_fuzzer-4933665846067200

Generate this asan error:

```
&nbsp; | ==291946==ERROR: AddressSanitizer: heap-use-after-free on address 0x622000015380 at pc 0x000000525dac bp 0x7fb1ddff69b0 sp 0x7fb1ddff6178
&nbsp; | WRITE of size 32 at 0x622000015380 thread T2
&nbsp; | SCARINESS: 55 (multi-byte-write-heap-use-after-free)
&nbsp; | #0 0x525dab in __asan_memset /src/llvm-project/compiler-rt/lib/asan/asan_interceptors_memintrinsics.cpp:26:3
&nbsp; | #1 0x2db1720 in Upsample&lt;2, 2&gt; libjxl/lib/jxl/dec_upsample.cc:106:3
&nbsp; | #2 0x2db1720 in jxl::N_AVX2::UpsampleRect(unsigned long, float const*, jxl::Plane&lt;float&gt; const&amp;, jxl::Rect const&amp;, jxl::Plane&lt;float&gt;*, jxl::Rect const&amp;, long, unsigned long, float*, unsigned long) libjxl/lib/jxl/dec_upsample.cc:245:7
&nbsp; | #3 0x2ddfce0 in UpsampleRect libjxl/lib/jxl/dec_upsample.cc:336:3
&nbsp; | #4 0x2ddfce0 in jxl::Upsampler::UpsampleRect(jxl::Image3&lt;float&gt; const&amp;, jxl::Rect const&amp;, jxl::Image3&lt;float&gt;*, jxl::Rect const&amp;, long, unsigned long, float*) const libjxl/lib/jxl/dec_upsample.cc:347:5
&nbsp; | #5 0x2d8ed4e in jxl::FinalizeImageRect(jxl::Image3&lt;float&gt;*, jxl::Rect const&amp;, std::__1::vector&lt;std::__1::pair&lt;jxl::Plane&lt;float&gt;*, jxl::Rect&gt;, std::__1::allocator&lt;std::__1::pair&lt;jxl::Plane&lt;float&gt;*, jxl::Rect&gt; &gt; &gt; const&amp;, jxl::PassesDecoderState*, unsigned long, jxl::ImageBundle*, jxl::Rect const&amp;) libjxl/lib/jxl/dec_reconstruct.cc:822:24
&nbsp; | #6 0x2d9ad27 in operator() libjxl/lib/jxl/dec_reconstruct.cc:1048:12
&nbsp; | #7 0x2d9ad27 in jxl::ThreadPool::RunCallState&lt;jxl::FinalizeFrameDecoding(jxl::ImageBundle*, jxl::PassesDecoderState*, jxl::ThreadPool*, bool, bool)::$_9, jxl::FinalizeFrameDecoding(jxl::ImageBundle*, jxl::PassesDecoderState*, jxl::ThreadPool*, bool, bool)::$_10&gt;::CallDataFunc(void*, unsigned int, unsigned long) libjxl/lib/jxl/base/data_parallel.h:88:14
&nbsp; | #8 0x3177779 in RunRange libjxl/lib/threads/thread_parallel_runner_internal.cc:137:7
...
```

<details>
 <summary>Full report</summary>

</details>

**Comment by eustas** [comment]
Looking into.

**Comment by eustas** [comment]
@veluca93 something is wrong with max size calculation storage is reserved for max width 288, but upsampling is requested for 512

**Comment by eustas** [comment]
Fixed by #111
